charts/spire/charts/spire-server/templates/statefulset.yaml

```diff
@@ -269,6 +274,12 @@ spec:
           secret:
             secretName: {{ include "spire-server.upstream-ca-secret" . }}
         {{- end }}
+        {{- if eq (.Values.upstreamAuthority.spire.enabled | toString) "true" }}
```

Comment:
This part should be controlled on if CSI drivers are being used, not if nested clusters are used

Reply:
Done